### DISCUSSION TITLE
open another nicegui application within a nicegui app

### DISCUSSION THREAD

**Discussion by TeeFiX42** [zauberzeug/nicegui#2873] [Q&A]
### Question

Hi,
 I have 1 central Webgui with a menu bar. Can it open other small (1-page) nicegui apps that can?
The webgui can take care of the authorization that allows the user to open the other small app. (The small apps have to be proxied via the central webgui app: they are not reachable directly.)
Is it possible to open a nicegui app in another nicegui app?
tnx

**Comment by rodja** [comment]
In principal this is possible. We do something similar with our [On Air](https://nicegui.io/documentation/section_configuration_deployment#nicegui_on_air) service (eg. proxying NiceGUI applications). But thats really tricky to get right and won't work directly for your use case. I suggest you use a reverse proxy like nginx or traefik to make all your NiceGUI Apps available to the users via different url paths. Then you can use `ui.element('iframe')` to embed pages from the sources into your main NiceGUI App.
